pygsti/tools/matrixtools.py

```diff
@@ -408,15 +453,33 @@ def independent_columns(m, initial_independent_cols=None, tol=1e-7):
 
 
 def pinv_of_matrix_with_orthogonal_columns(m):
-    """ TODO: docstring """
-    col_scaling = _np.sum(_np.abs(m)**2, axis=0)
+    """
+    Return the matrix "pinv_m" so m @ pinvm and pinv_m @ m are orthogonal projectors
+    onto subspaces of dimension rank(m).
+
+    Parameters
+    ----------
+    m : numpy.ndarray
+
+    Returns
+    ----------
+    pinv_m : numpy.ndarray
+    """
+    col_scaling = _np.linalg.norm(m, axis=0)**2
     m_with_scaled_cols = m.conj() * col_scaling[None, :]
     return m_with_scaled_cols.T
 
 
 def matrix_sign(m):
     """
-    The "sign" matrix of `m`
+    Compute the matrix s = sign(m). The eigenvectors of s are the same as those of m.
+    The eigenvalues of s are +/- 1, corresponding to the signs of m's eigenvalues.
+
+    It's straightforward to compute s when m is a normal operator. If m is not normal,
+    then the definition of s can be given in terms of m's Jordan form, and s
+    can be computed by (suitably post-processing) the Schur decomposition of m.
+
+    See https://nhigham.com/2020/12/15/what-is-the-matrix-sign-function/ for background.
```

Comment:
I made new codepaths that are more efficient for Hermitian matrices and normal matrices. (The asymptotics are the same, but they avoid running nested for-loops in Python.)